### DISCUSSION TITLE
usage in /server/api

### DISCUSSION THREAD

**Discussion by paulvonber** [Diizzayy/nuxt-graphql-client#326] [Q&A] [answered]
Is there any way to use composable's in server api's ?

**Comment by cfab** [comment]
Same question here !? is it possible ?

**Comment by DavidDeSloovere** [comment]
Also wondering the same.
I think this is important information that could be added to the docs.
My assumption is that you can't proxy to your own server routes, and thus would expose the auth to the client.

**Comment by Diizzayy** [comment, marked as answer]
@paulvonber @cfab @DavidDeSloovere This is already possible guys, though it's not reflected in the documentation. `Gql Functions` can be called via the server as seen below

```ts
export default eventHandler(async () => {
  const data = await GqlShips()

  return data
})
```

**Reply by cfab** [reply]
This is great ! For me it means easy caching with routeRules !

**Reply by DavidDeSloovere** [reply]
Yes! This is great. Thanks.

I'm not getting types on the server `.ts`, but they do work in the client `.vue` file.
Is this expected behaviour or something wrong on my machine?

EDIT:   Needed to manually import `import { GqlMovies } from '#gql';` in the server route. Now I get types.  Does auto-import not work in server routes in Nuxt?

**/pages/index.vue**
_auto imported_
![image](https://github.com/Diizzayy/nuxt-graphql-client/assets/352626/50bb6676-5b6e-4f34-a3dc-98d338516a38)

**/server/api/movies.get.ts**
_no import_
![image](https://github.com/Diizzayy/nuxt-graphql-client/assets/352626/9cfacbb4-7cd8-4bd0-86e6-72ed4a21845f)

_with manual import_
![image](https://github.com/Diizzayy/nuxt-graphql-client/assets/352626/8b746582-9d77-4ae5-b3b7-335794d4078e)

**Reply by devkon-at** [reply]
Manual import gets type hints working but i get the following error from nitro when running: 
Error: Could not load (...)nuxt/.nuxt/gql (imported by server/api/test.ts): EISDIR: illegal operation on a directory, read

Any idea on how to fix this?

**Reply by fouss** [reply]
can you use queries with fragment when you call server-side ?

**Reply by mulssoni** [reply]
> can you use queries with fragment when you call server-side ?

This is a very old post, but i'm having the same issue that fragments are not found. On client-side page route is working fine, but i also see the `Authorization` header in the network tab when SPA-navigating... How do people use this with sensitive non-personal api keys...?
